docs/task.md

```diff
@@ -0,0 +1,391 @@
+| [pull](#pull) | `bool` | Optional | Yes |
+
+* A [step](#step) must define either a [cmd](#cmd), [build](#build), or a [push](#push) property. It may not define more than one of the aforementioned properties.
+
+#### id
+
+The unique identifier for the [step](#step). Used as the name of the running container. Can be referenced across containers using this identifier as the tcp host and used in [when](#when).
+
+* Optional
+* Type: `string`
+* Cannot contain spaces.
+
+#### cmd
+
+Defines which container to run along with any additional command line arguments.
+
+Example:
+
+```yaml
+cmd: bash echo "hello world"
+```
+
+This runs a container called `bash` and tells it to run the `echo` command with `"Hello World"` as a parameter.
+
+* Optional
+* Type: `string`
+
+#### build
+
+Allows building containers.
+
+Example:
+
+```yaml
+build: -f Dockerfile -t acr-builder:v1 https://github.com/Azure/acr-builder.git
+```
+
+* Optional
+* Type: `string`
+
+#### entryPoint
+
+Sets the entry point of a container.
+
+* Optional
+* Type: `string`
+
+#### user
+
+Sets the username or UID of a container.
+
+* Optional
+* Type: `string`
+
+#### isolation
+
+Sets the isolation level of a container.
+
+* Optional
+* Type: `string`
+
+#### push
+
+Pushes the specified images to a container registry.
+
+Example:
+
+```yaml
+push: ["example.azurecr.io/acb:v1"]
+```
+
+* Optional
+* Type: `string[]`
+
+#### env
+
+Sets environment variables for the container during execution.
+
+Example:
+
+```yaml
+env: ["K1=V1", "K2=V2"]
+```
+
+* Optional
+* Type: `string[]`
+
+#### expose
+
+Exposes port(s) from the container.
+
+* Optional
+* Type: `string[]`
+
+#### ports
+
+Publishes port(s) from the container to the host.
+
+* Optional
+* Type: `string[]`
+
+#### when
+
+Describes when the step should get executed relative to other steps.
+If unspecified, steps will be ran sequentially.
+
+If `when: ["-"]` is specified, the container will immediately begin execution without any dependency on parent steps' execution.
+
+Examples:
+
+```yaml:
+# build is ran, followed by cmd sequentially.
+steps:
+  - build: -f Dockerfile . -t example
+  - cmd: example
+```
+
+```yaml:
+# step 1 and 2 are both executed simultaneously.
+steps:
+  - id: step_1
+    cmd: bash echo "1"
+
+  - id: step_2
+    cmd: bash echo "2"
+    when: ["-"]
+```
+
+```yaml:
+# step 1 and 2 are both executed simultaneously. Step 3 is executed after step 1 is completed.
+steps:
+  - id: step_1
+    cmd: bash echo "1"
+
+  - id: step_2
+    cmd: bash echo "2"
+    when: ["-"]
+
+  - id: step_3
+    cmd: bash echo "3"
+    when: ["step_1"]
+```
+
+* Optional
+* Type: `string[]`
+
+#### timeout
+
+The maximum execution time of a [step](#step) in seconds.
+
+* Optional
+* Type: `int`
+
+#### startDelay
+
+Defines the number of seconds to wait before executing the container.
+
+* Optional
+* Type: `int`
+
+#### retries
+
+The number of retries to attempt if a container fails its execution.
```

Comment:
Can you elaborate how we determine containers fails? eg, return code is not 0

Reply:
Done